openmdao/core/problem.py

```diff
@@ -2264,30 +2264,32 @@ def list_indep_vars(self, include_design_vars=True, options=None,
             raise RuntimeError("list_indep_vars requires that final_setup has been "
                                "run for the Problem.")
 
-        connections = model._conn_global_abs_in2out
         desvar_prom_names = model.get_design_vars(recurse=True,
                                                   use_prom_ivc=True,
                                                   get_sizes=False).keys()
+
         problem_indep_vars = []
         indep_var_names = set()
 
         default_col_names = ['name', 'units', 'val']
         col_names = default_col_names + ([] if options is None else options)
 
-        for target, meta in model._var_allprocs_abs2meta['input'].items():
-            src = connections[target]
-            smeta = model._var_allprocs_abs2meta['output'][src]
-            src_is_ivc = 'openmdao:indep_var' in smeta['tags']
-            input_name = model._var_allprocs_abs2prom['input'][target]
+        abs2meta = model._var_allprocs_abs2meta['output']
+        prom2abs = self.model._var_allprocs_prom2abs_list['input']
+
+        prom2src = {prom: model.get_source(prom) for prom in prom2abs.keys()
+                    if 'openmdao:indep_var' in abs2meta[model.get_source(prom)]['tags']}
 
-            smeta = {key: val for key, val in smeta.items() if key in col_names}
-            smeta['val'] = self.get_val(input_name)
+        for prom, src in prom2src.items():
+            name = prom if '_auto_ivc' in src else src
```

Comment:
`name = prom if src.startswith('_auto_ivc.') else src` would work better in the highly unlikely event that src contained the string '_auto_ivc' but wasn't actually the path of a variable contained in the _auto_ivc component.

Reply:
added `src.startswith('_auto_ivc.')` to the check